en/advanced/ai.md

```diff
@@ -0,0 +1,79 @@
+
+## Instruction
+
+**Type**: string
+
+**Requirements**: not empty
+
+An instruction (also known as "system message") in a Large Language Models (LLMs) sets the tone and rules for the conversation. Think of it as instructions given to the AI before it starts interacting with a user. It guides the AI on how to respond, ensuring it stays on topic and behaves appropriately. For example, a system message might tell the AI to be formal, concise, or provide detailed explanations. This helps the AI provide more relevant and useful answers tailored to the specific needs of the user.
+
+## Message window size
+
+**Type**: integer
+
+**Requirements**: > 0
+
+The "message window size" in a Large Language Model (LLM) refers to the number of recent messages or interactions that the model remembers during a conversation. This parameter determines how much context the LLM considers when generating responses.
+
+In LLMs, the context window is like a memory that helps the model understand the conversation flow and provide relevant responses. However, due to computational limits, LLMs cannot store unlimited amounts of chat history. Therefore, the "message window size" specifies the maximum number of previous messages or turns in a conversation that the LLM retains. This allows the LLM to focus on the most recent interactions, improving response accuracy without overwhelming the system with unnecessary data.
+
+## Document splitter chunk size
+
+**Type**: integer
+
+**Requirements**: > 0
+
+The "chunk size" parameter in document splitting refers to the size of segments into which linked files are divided for processing by AI models. When dealing with linked files, such as PDF files, they are segmented into smaller chunks based on this parameter. Each segment typically contains a specified number of words or characters, ensuring manageable units for analysis and generating answers.
+
+These segments are then passed to the AI model for processing. This approach helps optimize performance by breaking down large documents into smaller, more digestible parts, enabling more efficient handling and analysis by the AI.
+
+## Document splitter chunk overlap
+
+**Type**: integer
+
+**Requirements**: > 0 && < chunk size
+
+The "chunk overlap" parameter determines how much text from adjacent chunks is shared when dividing linked files into segments. This overlap is measured in characters and ensures continuity and context across segmented chunks. By sharing a specified amount of text between adjacent segments, typically at the beginning and/or end of each chunk, the AI model can maintain coherence and understanding of the content across segmented parts. This approach helps enhance the accuracy and relevance of responses generated by the AI from the segmented content.
+
+## Retrieval augmented generation maximum results count
```

Comment:
"Maximum results count" sounds complicated. I prefer naming this "Maximum number of chunks" or "Maximum number of snippets".